Rama feature cotitular

edrosolis opened the pull request:
-Pull request cotitular
-Edgardo solis

JuanjoDBZ committed:
-Se creó módulo para cotitulares.
-Se creó vista para cotitulares.
-Se crea Entity SABPaymentInformationRequestEntity.
-Se crea Entity SABTypePaysMembershipsInformationRequestEntity.
-Se cambia parametro "parametersCreateToken" dictionary por entity.
-Se agrega redondeo a contenedores y cambio de color SABRealizaTuPagoVC.
-Crea Edgardo Solís.

JuanjoDBZ committed:
-Se crea modulo  SABCoOwner
-Se crea modulo PlaceHolderCollectionView
-Se crea modulo CoOwnerCollectionView
-Se crea modulo NewCoOwner
-Se agrega Assets Group

JuanjoDBZ committed:
-Se crea modulo SABNewCoOwner
-Se crea lógica para cargar los datos del nuevo cotitular en el collectionView

JuanjoDBZ committed:
-Se modifica lógica en la calle SABCoOwner para mostrar información en el collectionview

JuanjoDBZ committed:
-Se elimino lógica para pasar los datos de la vista SABNewCoOwner
-Se agrega campos para el llamado de servicio de SABNewCoOwner

JuanjoDBZ committed:
-Se agrega comentarios a funciones y variables de las modelos SABCoOwner y SABNewCoOwner
-Se cambio la vista que va a cargar la aplicación

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/Interactor/SABCoOwnerInteractor.swift:
```diff
@@ -0,0 +1,16 @@
+//
+//  Created by IsitaFS003 on 21/07/21.
+//  
+//
```
Eliminar espacios @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/Presenter/SABCoOwnerPresenter.swift:
```diff
@@ -0,0 +1,21 @@
+//  Created by IsitaFS003 on 21/07/21.
+//  
+//
+
```
Eliminar espacios @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/Protocol/SABCoOwnerProtocol.swift:
```diff
@@ -0,0 +1,31 @@
+//  Created by IsitaFS003 on 21/07/21.
+//  
+//
+
```
Eliminar espacios @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/Router/SABCoOwnerRouter.swift:
```diff
@@ -0,0 +1,35 @@
+//
+//  Created by IsitaFS003 on 21/07/21.
+//  
+//
```
Eliminar espacios @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/CollectionViewCell/SABCoOwnerCollectionViewCell.swift:
```diff
@@ -0,0 +1,17 @@
+//
+//  Created by IsitaFS003 on 21/07/21.
+//
+
```
Si la función override func awakeFromNib() no se usa eliminarla @edrosolis . Porque esta celda no tiene nada ? apenas la vas a utilizar ?

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/SABCoOwnerVC.swift:
```diff
@@ -0,0 +1,37 @@
+//  
+//
+import Foundation
+import UIKit
```
Eliminar espacios @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/SABCoOwnerVC.swift:
```diff
@@ -0,0 +1,37 @@
+
+/// SABCoOwnerVC
+class SABCoOwnerVC: UIViewController {
+    /// Objeto CollectionView
```
Eliminar espacios y comentar variables @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/SABCoOwnerVC.swift:
```diff
@@ -0,0 +1,37 @@
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+    }
```
Comentar funciones @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/SABCoOwnerVC.swift:
```diff
@@ -0,0 +1,37 @@
+extension SABCoOwnerVC: SABCoOwnerViewProtocol {
+}
+extension SABCoOwnerVC: UICollectionViewDelegate,UICollectionViewDataSource {
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
```
Comentar funciones @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABLogin/Router/SABLoginRouter.swift:
```diff
@@ -31,18 +31,19 @@ extension SABLoginRouter: SABLoginRouterProtocol {
-        factura.tabBarItem.image = UIImage(named: "factura")
-        compras.tabBarItem.image = UIImage(named: "house")
-        cotitular.tabBarItem.image = UIImage(named: "cotitular")
+        let homeVC = UINavigationController(rootViewController: SABTabBarHomeRouter().view)
```
Comentar variables @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABRegister/MakePayment/Entity/SABPaymentInformationRequestEntity.swift:
```diff
@@ -0,0 +1,22 @@
+    var expirationMonth: String = ""
+    /// CVV
+    var cvv: String = ""
+    /// customerId
```
Comentar variable de manera mas especifica y no igual que su nombre.

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABRegister/MakePayment/Entity/SABPaymentInformationRequestEntity.swift:
```diff
@@ -0,0 +1,22 @@
+//
+import Foundation
+struct DataMakePayment {
+    /// Numero de tarjeta
```
Revisar acentos @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABRegister/MakePayment/RemoteData/SABRealizaTuPagoRemoteDataManagerInputProtocol.swift:
```diff
@@ -79,23 +93,29 @@ class SABRealizaTuPagoRemoteDataManagerInputProtocol: NSObject {
+        dictionarycvvCardForm = TokenCard.cvv
+        dictionarynameForm = TokenCard.holderName
+        customerId = TokenCard.customerId
+//        for item in TokenCard {
```
Si no se va usar eliminar lo comentado @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABRegister/MakePayment/RemoteData/SABRealizaTuPagoRemoteDataManagerInputProtocol.swift:
```diff
@@ -7,20 +7,34 @@
 import OpenpayKit
 class SABRealizaTuPagoRemoteDataManagerInputProtocol: NSObject {
-    /// Variables para los consumos de servicios
+    /// MERCHANT_ID
```
Porque estan hardCode esos datos ? @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABRegister/MakePayment/RemoteData/SABRealizaTuPagoRemoteDataManagerInputProtocol.swift:
```diff
@@ -79,23 +93,29 @@ class SABRealizaTuPagoRemoteDataManagerInputProtocol: NSObject {
         var dictionaryexpMonthCardForm = ""
         var dictionarycvvCardForm = ""
         var dictionarynameForm = ""
-        for item in TokenCard {
```
No pudes usar forceWrap usa los if let @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABRegister/MakePayment/RemoteData/SABRealizaTuPagoRemoteDataManagerInputProtocol.swift:
```diff
@@ -79,23 +93,29 @@ class SABRealizaTuPagoRemoteDataManagerInputProtocol: NSObject {
         var dictionaryexpMonthCardForm = ""
```
Comenta las variables @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABRegister/MakePayment/RemoteData/SABRealizaTuPagoRemoteDataManagerInputProtocol.swift:
```diff
@@ -79,23 +93,29 @@ class SABRealizaTuPagoRemoteDataManagerInputProtocol: NSObject {
+//        }
         let card = TokenizeCardRequest(cardNumber: dictionaryCardNumberForm,holderName:dictionarynameForm, expirationYear: dictionaryexpYearCardForm, expirationMonth: dictionaryexpMonthCardForm, cvv2: dictionarycvvCardForm)
         openpay.tokenizeCard(card: card) { (OPToken) in
             print("token de tarjeta")
```
Eliminar el print @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABRegister/MakePayment/View/SABRealizaTuPagoVC.swift:
```diff
@@ -5,35 +5,51 @@
-    /// Declaracion de variables  para logica de integracion de datos para consumo de servicios 
+    /// Textfield nombre del usuario
+    @IBOutlet weak var txtName: UITextField!
+    /// Textfield numero de tarjeta
```
Revisar acentos @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/CoOwnerCollectionViewCell/Interactor/CoOwnerCollectionViewCellInteractor.swift:
```diff
@@ -0,0 +1,13 @@
+//
+//  Created by IsitaFS003 on 23/07/21.
+//
+
```
Eliminar espacios @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/CoOwnerCollectionViewCell/Presenter/CoOwnerCollectionViewCellPresenter.swift:
```diff
@@ -0,0 +1,19 @@
+//
+//  Created by IsitaFS003 on 23/07/21.
+//
+
```
Eliminar espacios @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/CoOwnerCollectionViewCell/View/CoOwnerCollectionViewCell.swift:
```diff
@@ -0,0 +1,25 @@
+//
+//  Created by IsitaFS003 on 22/07/21.
+//
+
```
Eliminar espacios @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/CoOwnerCollectionViewCell/View/CoOwnerCollectionViewCell.swift:
```diff
@@ -0,0 +1,25 @@
+
+    var presenter: CoOwnerCollectionViewCellPresenterProtocol?
+    @IBOutlet weak var buttonGroup: UIButton!
+    override func awakeFromNib() {
```
Si no se usa la función eliminarla, comentar el outlet, eliminar espacios @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/CoOwnerCollectionViewCell/View/CoOwnerCollectionViewCell.swift:
```diff
@@ -0,0 +1,25 @@
+    }
+
+    @IBAction func buttonNewGroup(_ sender: UIButton) {
+        print("entro")
```
Eliminar print @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/NewCoOwner/SABNewCoOwnerInteractor.swift:
```diff
@@ -0,0 +1,16 @@
+//  Created by IsitaFS003 on 26/07/21.
+//  
+//
+
```
Eliminar espacios @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/NewCoOwner/SABNewCoOwnerPresenter.swift:
```diff
@@ -0,0 +1,21 @@
+//  Created by IsitaFS003 on 26/07/21.
+//  
+//
+
```
Eliminar espacios @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/NewCoOwner/SABNewCoOwnerRouter.swift:
```diff
@@ -0,0 +1,29 @@
+//
+//  Created by IsitaFS003 on 26/07/21.
+//  
+//
```
Eliminar espacios @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/NewCoOwner/SABNewCoOwnerVC.swift:
```diff
@@ -0,0 +1,22 @@
+//  Created by IsitaFS003 on 26/07/21.
+//  
+//
+
```
Eliminar espacios @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/NewCoOwner/SABNewCoOwnerVC.swift:
```diff
@@ -0,0 +1,22 @@
+import UIKit
+
+class SABNewCoOwnerVC: UIViewController {
+
```
Eliminar espacios @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/PlaceHolderCollectionViewCell/Router/PlaceHolderCollectionViewCellRouter.swift:
```diff
@@ -0,0 +1,34 @@
+//  ShipAndBox
+//
+//  Created by IsitaFS003 on 23/07/21.
+//
```
Estos archivos del CellRouter son los que no se están usando ? si son esos hay que eliminarlos @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/PlaceHolderCollectionViewCell/View/PlaceHolderCollectionViewCell.swift:
```diff
@@ -0,0 +1,32 @@
+//
+import UIKit
+class PlaceHolderCollectionViewCell: UICollectionViewCell {
+    var presenter: PlaceHolderCollectionViewCellPresenterProtocol?
```
Comentar outlet, eliminar variable presenter @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/PlaceHolderCollectionViewCell/View/PlaceHolderCollectionViewCell.swift:
```diff
@@ -0,0 +1,32 @@
+        super.awakeFromNib()
+        setUpView()
+    }
+    func setUpView() {
```
Comentar función @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/PlaceHolderCollectionViewCell/View/PlaceHolderCollectionViewCell.swift:
```diff
@@ -0,0 +1,32 @@
+        yourViewBorder.path = UIBezierPath(rect: viewConteiner.bounds).cgPath
+        viewConteiner.layer.addSublayer(yourViewBorder)
+    }
+    @IBAction func buttonAddCoOwner(_ sender: UIButton) {
```
Comentar función @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/SABCoOwnerVC.swift:
```diff
@@ -7,31 +7,64 @@
     @IBOutlet weak var collectionCoOwner: UICollectionView!
     var arrayCoOwner: Array = [""]
-    
+    var arrayPlaceHolderOwner: Array = [""]    
```
Creo que hay otra manera de declarar un array vacío @edrosolis , comentar para que es ese array

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/SABCoOwnerVC.swift:
```diff
@@ -7,31 +7,64 @@
-    
+    var arrayPlaceHolderOwner: Array = [""]    
     var presenter: SABCoOwnerPresenterProtocol?
 
```
Eliminar espacios @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/SABCoOwnerVC.swift:
```diff
@@ -7,31 +7,64 @@
     }
     func setUpCollectionView() {
-        self.collectionCoOwner.register(UINib(nibName: "CustomCollectionViewCell", bundle: nil), forCellWithReuseIdentifier: "customReuseIdentifier")
+        collectionCoOwner.dataSource = self
```
Comentar función @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/SABCoOwnerVC.swift:
```diff
@@ -7,31 +7,64 @@
+    func numberOfSections(in collectionView: UICollectionView) -> Int {
         return arrayCoOwner.count
+      }
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
```
Comentar todos los delegados de manera sencilla @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/SABCoOwnerVC.swift:
```diff
@@ -7,31 +7,64 @@
     func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
-        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "SABCoOwnerCollectionViewCell", for: indexPath) as! SABCoOwnerCollectionViewCell
-        return cell
+        var cell:UICollectionViewCell? = nil
```
Comentar variable y para que se usa brevemente @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/SABCoOwnerVC.swift:
```diff
@@ -7,31 +7,64 @@
+        var cell:UICollectionViewCell? = nil
+        let section = indexPath.section
+
+        if section == 0 {
```
Busca la manera de quitar los forcewrap de la celda después puede tronar la app si por alguna razón no viene el cell @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SceneDelegate.swift:
```diff
@@ -20,7 +20,9 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
                 window = UIWindow(frame: windowScene.coordinateSpace.bounds)
                 window?.windowScene = windowScene
         let login = SABLoginRouter()
```
Si vas a subir algo para hacer un cambio tienes que regresar todo a su estado original @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/Router/SABCoOwnerRouter.swift:
```diff
@@ -26,5 +26,16 @@ class SABCoOwnerRouter {
+}
+extension SABCoOwnerRouter: NewPersonProtocol{
+    func saveNewPerson(persona: newPersonModel) {
+        print("dededededed")
```
Eliminar prints @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/CollectionViewCell/CoOwnerCollectionViewCell/CoOwnerCollectionViewCell.swift:
```diff
@@ -9,7 +9,9 @@ import UIKit
 class CoOwnerCollectionViewCell: UICollectionViewCell {
 
-    var presenter: CoOwnerCollectionViewCellPresenterProtocol?
+    @IBOutlet weak var labelName: UILabel!
```
Comentar outlets, eliminar función si no se utiliza @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/CollectionViewCell/CoOwnerCollectionViewCell/CoOwnerCollectionViewCell.swift:
```diff
@@ -20,6 +22,4 @@ class CoOwnerCollectionViewCell: UICollectionViewCell {
         print("entro")
```
Eliminar print @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/CollectionViewCell/PlaceHolderCollectionViewCell/PlaceHolderCollectionViewCell.swift:
```diff
@@ -6,7 +6,6 @@
 import UIKit
 class PlaceHolderCollectionViewCell: UICollectionViewCell {
-    var presenter: PlaceHolderCollectionViewCellPresenterProtocol?
     @IBOutlet weak var viewConteiner: UIView!
```
Comentar outlet @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/CollectionViewCell/PlaceHolderCollectionViewCell/PlaceHolderCollectionViewCell.swift:
```diff
@@ -6,7 +6,6 @@
-    var presenter: PlaceHolderCollectionViewCellPresenterProtocol?
     @IBOutlet weak var viewConteiner: UIView!
     override func awakeFromNib() {
         super.awakeFromNib()
```
Eliminar clase comentada @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/SABCoOwnerVC.swift:
```diff
@@ -11,8 +11,9 @@ import UIKit
     @IBOutlet weak var collectionCoOwner: UICollectionView!
-    var arrayCoOwner: Array = [""]
-    var arrayPlaceHolderOwner: Array = [""]    
+    var arrayCoOwner: Array<Any> = ["",""]
```
Comentar variables, investigar si hay otra manera de declararlos vacíos @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/SABCoOwnerVC.swift:
```diff
@@ -29,6 +30,11 @@ class SABCoOwnerVC: UIViewController {
+    func prueba2(newPerson: newPersonModel) {
+        arrayNewCoOwner = newPerson
+        collectionCoOwner.reloadData()
+
```
Eliminar espacio y comentar función @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/SABCoOwnerVC.swift:
```diff
@@ -29,6 +30,11 @@ class SABCoOwnerVC: UIViewController {
+    }
 }
 extension SABCoOwnerVC: UICollectionViewDelegate,UICollectionViewDataSource {
     func numberOfSections(in collectionView: UICollectionView) -> Int {
```
Comentar función.

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/SABCoOwnerVC.swift:
```diff
@@ -38,33 +44,40 @@ extension SABCoOwnerVC: UICollectionViewDelegate,UICollectionViewDataSource {
         }
+        return 0
     }
     func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
```
Comentar todos los delegados de manera simple @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/SABCoOwnerVC.swift:
```diff
@@ -38,33 +44,40 @@ extension SABCoOwnerVC: UICollectionViewDelegate,UICollectionViewDataSource {
     }
-    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, minimumLineSpacingForSectionAt section: Int) -> CGFloat {
-        return 5
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
```
Eliminar todos los prints @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABNewCoOwner/Entity/SABPersonModel.swift:
```diff
@@ -0,0 +1,13 @@
+//
+//  Created by IsitaFS003 on 27/07/21.
+//
+
```
Eliminar espacio y comentar variables @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABNewCoOwner/Router/SABNewCoOwnerRouter.swift:
```diff
@@ -2,7 +2,7 @@
+//  Created by IsitaFS003 on 27/07/21.
 //  
 //
 
```
Eliminar espacios @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABNewCoOwner/View/SABNewCoOwnerVC.swift:
```diff
@@ -0,0 +1,48 @@
+//  Created by IsitaFS003 on 27/07/21.
+//  
+//
+
```
Eliminar espacios @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABNewCoOwner/View/SABNewCoOwnerVC.swift:
```diff
@@ -0,0 +1,48 @@
+
+class SABNewCoOwnerVC: UIViewController {
+
+    @IBOutlet weak var txtFullName: UITextField!
```
Comentar outlets @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABNewCoOwner/View/SABNewCoOwnerVC.swift:
```diff
@@ -0,0 +1,48 @@
+
+import Foundation
+import UIKit
+protocol NewPersonProtocol: class {
```
Comentar protocolo si se utiliza, si no se utiliza eliminarlo @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABNewCoOwner/View/SABNewCoOwnerVC.swift:
```diff
@@ -0,0 +1,48 @@
+    var delegate:NewPersonProtocol?
+    var arrayPerson = [String]()
+    var newPerson: newPersonModel = newPersonModel()
+
```
Eliminar espacios @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABNewCoOwner/View/SABNewCoOwnerVC.swift:
```diff
@@ -0,0 +1,48 @@
+    override func viewDidLoad() {
+        super.viewDidLoad()
+    }
+    
```
Eliminar espacios @edrosolis, comentar funciones y eliminar todo código comentado.

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/SABCoOwnerVC.swift:
```diff
@@ -11,9 +11,10 @@ import UIKit
-    var arrayCoOwner: Array<Any> = ["",""]
-    var arrayPlaceHolderOwner: Array = ["",""]
-    var arrayNewCoOwner: newPersonModel = newPersonModel()
+    var arrayCoOwner: Array<Any> = []
```
Comentar outlets, comentar variables, ver si hay otra forma de declarar los arreglos vacíos @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/SABCoOwnerVC.swift:
```diff
@@ -31,53 +32,73 @@ class SABCoOwnerVC: UIViewController {
 ///Protocolo para recibir datos de presenter.
 extension SABCoOwnerVC: SABCoOwnerViewProtocol {
     func prueba2(newPerson: newPersonModel) {
```
Cambiar nombre de la función, comentar la función @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/SABCoOwnerVC.swift:
```diff
@@ -31,53 +32,73 @@ class SABCoOwnerVC: UIViewController {
+        modelNewCoOwner.append(newPerson)
+        arrayCoOwner.append(newPerson)
         collectionCoOwner.reloadData()
-
```
Eliminar espacios @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/SABCoOwnerVC.swift:
```diff
@@ -31,53 +32,73 @@ class SABCoOwnerVC: UIViewController {
     }
 }
 extension SABCoOwnerVC: UICollectionViewDelegate,UICollectionViewDataSource {
     func numberOfSections(in collectionView: UICollectionView) -> Int {
```
Comentar la función @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/SABCoOwnerVC.swift:
```diff
@@ -31,53 +32,73 @@ class SABCoOwnerVC: UIViewController {
+            }
         }
         return 0
     }
```
Comentar los delegados de manera simple @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/SABCoOwnerVC.swift:
```diff
@@ -31,53 +32,73 @@ class SABCoOwnerVC: UIViewController {
+            if arrayCoOwner.count == 0 {
+               let cellPlaceHolder = collectionView.dequeueReusableCell(withReuseIdentifier:"PlaceHolderCollectionViewCell", for: indexPath)
+                return cellPlaceHolder
+            }else{
```
espacio entre el else y el curly bracket @edrosolis

JuanjoDBZ commented on ShipAndBox/ShipAndBox/SABCoOwner/View/SABCoOwnerVC.swift:
```diff
@@ -31,53 +32,73 @@ class SABCoOwnerVC: UIViewController {
+                return cellPlaceHolder
+            }else{
+                let cellCo = collectionView.dequeueReusableCell(withReuseIdentifier:"CoOwnerCollectionViewCell", for: indexPath) as! CoOwnerCollectionViewCell
+                let coOwnerList = modelNewCoOwner[indexPath.row]
```
Buscar la manera de eliminar el forcewrap y comentar variables

JuanjoDBZ reviewed (changes requested):
Hacen falta cambios. @edrosolis

JuanjoDBZ committed:
-Se borraron espacios en la clases
-Se  agregan comentarios en variables, IBOutlet y funciones
-Se  borraron print en las clases
-Edgardo solís

JuanjoDBZ committed:
-Se cambia nombre de funciones

JuanjoDBZ committed:
-Se elimina el forcewrap en el cellForItemAt en la clase SABCoOwnerVC
